Library/Homebrew/download_strategy.rb

```diff
@@ -555,24 +557,93 @@ def curl(*args, **options)
   end
 end
 
-# Strategy for downloading a file from an GitHub Packages URL.
+# Strategy for downloading a file from a Docker registry.
 #
 # @api public
-class CurlGitHubPackagesDownloadStrategy < CurlDownloadStrategy
-  attr_writer :resolved_basename
+class DockerRegistryDownloadStrategy < CurlDownloadStrategy
+  attr_writer :mimetype, :resolved_basename
+
+  # Will store tokens for re-use
+  @@bearer_tokens = {} # rubocop:disable Style/ClassVars
```

Comment:
Use an instance instead of class variable here. `extend Cachable` may be of interest/help here.

Reply:
Won't each `Resource` have its own instance of `DockerRegistryDownloadStrategy`?

Reply:
Yes. I think a class variable is a bad call here, better to figure out another way of e.g. passing in a hash that could be shared.